src/TorchSharp/JIT/ScriptModule.cs

```diff
@@ -373,13 +367,17 @@ public unsafe override Tensor forward(Tensor x, Tensor y, Tensor z)
                 /// <returns></returns>
                 public unsafe Tensor forward(Tensor x, Tensor y, Tensor z, params Tensor[] tensors)
                 {
-                    using (var parray = new PinnedArray<IntPtr>()) {
-                        IntPtr tensorRefs = parray.CreateArray(new[] { x, y, z }.Concat(tensors).Select(t => t.Handle).ToArray());
-                        var res = THSJIT_Module_forward(handle, tensorRefs, parray.Array.Length);
-                        if (res == IntPtr.Zero)
-                            CheckForErrors();
-                        return new Tensor(res);
-                    }
+                    var count = 3 + tensors.Length;
+                    var tensorRefs = stackalloc IntPtr [count];
```

Comment:
Can the `count` be large number? I am asking because the stack space is limited.

Reply:
It's the number of arguments to a function, so it would be very unlikely. I can check for a large number and switch to 'new', I guess.

Reply:
Updated code.